lib/src/models/rover/controller.dart

```diff
@@ -62,7 +76,8 @@ class Controller extends Model {
 
 	/// Sends a command over the network or over Serial.
 	Future<void> sendMessage(Message message) async {
-		if (models.serial.isConnected) {
+		if (models.serial.isConnected &&
+	(teensyCommands.containsKey(message.messageName))) {
```

Comment:
This is saying "if serial is connected and the message we're sending is IN the map of all commands". That should always be true, since all the messages we're sending are in that map. You want to check which Teensy that kind of command is meant for, and check if that's the Teensy we're currently connected to.